src/main/java/com/google/devtools/build/lib/packages/TargetUtils.java

```diff
@@ -59,6 +62,17 @@ public boolean apply(String tag) {
     }
   };
 
+  // Not all the target level tags are propagated as execution requirements to the actions, because
+  // we do not want to propagate custom user's tags or create potential conflicts
+  // with execution requirements declared on the rules.
+  // See https://github.com/bazelbuild/bazel/issues/7766 for details.
+  private static final Predicate<String> TAGS_PROPAGATED_TO_EXEC_INFO =
```

Comment:
we could, but this is not what we've agreed on in the initial design.

Reply:
Independent what's in the design doc, let's not do it if it makes no sense? :)

Reply:
I think it makes sense to keep them separate because they serve different purposes: `LEGAL_EXEC_INFO_KEYS` specifies the universe of execution info keys, and `TAGS_PROPAGATED_TO_EXEC_INFO` specifies the specific tags that can be propagated. 

`TAGS_PROPAGATED_TO_EXEC_INFO` is a subset of  `LEGAL_EXEC_INFO_KEYS`

In `getExecutionInfoFromTags`, we only propagate tags that satisfy `TAGS_PROPAGATED_TO_EXEC_INFO`, irrespective of whether they satisfy `LEGAL_EXEC_INFO_KEYS`; I don't see how we'd do this check if the two predicates were combined.